pkg/controllers/nodeclass/validation.go

```diff
@@ -40,6 +65,117 @@ func (n Validation) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (
 			fmt.Sprintf("%q tag does not pass tag validation requirements", offendingTag))
 		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("%q tag does not pass tag validation requirements", offendingTag))
 	}
+	//Auth Validation
+	amis, err := n.amiProvider.List(ctx, nodeClass)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if nodeClass.StatusConditions().Get(v1.ConditionTypeSubnetsReady).IsFalse() || nodeClass.StatusConditions().Get(v1.ConditionTypeAMIsReady).IsFalse() {
+		return reconcile.Result{}, nil
+	}
+	nodeClaim := &karpv1.NodeClaim{
+		Spec: karpv1.NodeClaimSpec{
+			NodeClassRef: &karpv1.NodeClassReference{
+				Name: nodeClass.ObjectMeta.Name,
+			},
+		},
+		Status: karpv1.NodeClaimStatus{
+			ImageID: amis[0].AmiID,
+		},
+	}
+	tags, err := utils.GetTags(ctx, nodeClass, nodeClaim, options.FromContext(ctx).ClusterName)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	var errs []error
+
+	createFleetInput := n.instanceProvider.GetCreateFleetInput(nodeClass, string(karpv1.CapacityTypeOnDemand), tags, mockLaunchTemplateConfig(), true)
+
+	if _, err := n.ec2api.CreateFleet(ctx, createFleetInput); awserrors.IsNotDryRunError(err) {
+		errs = append(errs, fmt.Errorf("create fleet %w", err))
+	}
+
+	createLaunchTemplateInput := n.launchTemplateProvider.GetCreateLaunchTemplateInput(mockOptions(*nodeClaim, nodeClass, tags), corev1.IPv4Protocol, "", true)
+
+	if _, err := n.ec2api.CreateLaunchTemplate(ctx, createLaunchTemplateInput); awserrors.IsNotDryRunError(err) {
+		errs = append(errs, fmt.Errorf("create launch template %w", err))
+	}
+
+	imageOutput, err := n.ec2api.DescribeImages(ctx, &ec2.DescribeImagesInput{
```

Comment:
Why do we need to get the AMIs again? Can we just use the values that are retrieved from the AMI provider?